bridge/setu/listener/rootchain_selfheal.go

```diff
@@ -199,32 +208,26 @@ func (rl *RootChainListener) processStateSynced(ctx context.Context) {
 	}
 }
 
-func (rl *RootChainListener) processEvent(ctx context.Context, event types.Log) (bool, error) {
-	// Check existence of topics beforehand and ignore if no topic exists
-	// (TODO): Identify issue of empty events: See Jira POS-818
-	if len(event.Topics) == 0 {
-		return true, nil
-	}
-
-	blockTime, err := rl.contractConnector.GetMainChainBlockTime(ctx, event.BlockNumber)
+func (rl *RootChainListener) processEvent(ctx context.Context, vLog *types.Log) (bool, error) {
+	blockTime, err := rl.contractConnector.GetMainChainBlockTime(ctx, vLog.BlockNumber)
 	if err != nil {
 		rl.Logger.Error("Unable to get block time", "error", err)
 		return false, err
 	}
 
 	if time.Since(blockTime) < helper.GetConfig().SHMaxDepthDuration {
-		rl.Logger.Info("Block time is less than an hour, skipping state sync")
+		rl.Logger.Info("Block time is less than max time depth, skipping event")
 		return true, err
 	}
 
-	topic := event.Topics[0].Bytes()
+	topic := vLog.Topics[0].Bytes()
```

Comment:
I've seen before that vLog i.e. event instance was empty in some cases (mentioned in one of the comment above). Is it still the case or we can safely skip that check. Just a minor concern because this statement here will panic if it's empty.

Reply:
It's safe now, I found the reason why it was empty earlier and fixed that.